pkg/util/tail/tail_test.go

```diff
@@ -18,10 +18,58 @@ package tail
 
 import (
 	"bytes"
+	"io/ioutil"
+	"os"
 	"strings"
 	"testing"
 )
 
+func TestReadAtMost(t *testing.T) {
```

Comment:
can you reconstruct the test as a table drive test? so that we do not need to repeat the same code logic 3 times.

https://github.com/golang/go/wiki/TableDrivenTests